packages/core/src/Serialization/TransactionBody/TransactionBody.ts

```diff
@@ -95,14 +97,9 @@ export class TransactionBody {
     //   }
     writer.writeStartMap(this.#getMapSize());
 
-    if (this.#inputs && this.#inputs.length > 0) {
+    if (this.#inputs && this.#inputs.values().length > 0) {
```

Comment:
Maybe you can add a size property to the CborSet itself, so you can simply call `this.#inputs.size()`

Reply:
Done in [here](https://github.com/input-output-hk/cardano-js-sdk/commit/bb6334982335b27733c25ea0b2dead9f5829bf99) and [here](https://github.com/input-output-hk/cardano-js-sdk/commit/0eee790392fe6336da9b5dc62f5f4b5ce1655a82)